### PR TITLE
Allow variable and function with the same name in static blocks

### PR DESCRIPTION
**Pull request by Yokubjon-J**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes [#14257](https://github.com/babel/babel/issues/14257)  
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Allow a variable and a function to have a same name in a static block

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14344"></a>

### CODE REVIEW

**Comment by Yokubjon-J** [comment]
I wonder why babel bot shows another issue (not about static block) in REPL...

**Comment by nicolo-ribaudo** [comment]
You can edit the REPL contents to whatever you want :wink:

**Review by nicolo-ribaudo — APPROVED** [review]
The CI failure is not related to this PR, it's a flaky test that sometimes fails.

**Review by JLHwung — COMMENTED** [review]
static block is an ES2022 feature, can you move the test case to es2022? There is a static block folder over there. I think it suffices to add only one test case.

**Comment by Yokubjon-J** [comment]
I have created a new subdirectory: `...\es2022\class-private-names-duplicated\static-block-duplicate-name`. The test has passed.

File: packages/babel-runtime-corejs3/helpers/esm/possibleConstructorReturn.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import _typeof from "./typeof.js";
+import _typeof from "@babel/runtime-corejs3/helpers/typeof";
```
Comment (nicolo-ribaudo) [inline packages/babel-runtime-corejs3/helpers/esm/possibleConstructorReturn.js:1]: Are you using windows? I just realized that this unwanted change (I'll revert it again) might be a bug in out build process.

Comment (Yokubjon-J) [reply packages/babel-runtime-corejs3/helpers/esm/possibleConstructorReturn.js:1]: yes, i am using windows! I have reverted it just now!

Comment (Yokubjon-J) [reply packages/babel-runtime-corejs3/helpers/esm/possibleConstructorReturn.js:1]: now i am rerunning `make bootstrap`

File: packages/babel-parser/test/fixtures/es2022/class-private-names-duplicated/static-block-duplicate-name/input.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class X {
```
Comment (JLHwung) [inline packages/babel-parser/test/fixtures/es2022/class-private-names-duplicated/static-block-duplicate-name/input.js:1]: We can remove the other two tests in `es2015`, only the `es2022` one is sufficient.

Not related to this PR: I realize that the `static-block` test is still in the `experimental` folder, we should have moved it to the `es2022`.

**Review by JLHwung — APPROVED** [review]
Thanks!

**Pull request merged**